pytorch_lightning/trainer/trainer.py

```diff
@@ -1862,8 +1866,19 @@ def reset_train_dataloader(self, model: Optional["pl.LightningModule"] = None) -
                 category=PossibleUserWarning,
             )
 
-        # store epoch of dataloader reset for reload_dataloaders_every_n_epochs
-        self._last_train_dl_reload_epoch = self.current_epoch
+        if (
+            self.num_training_batches == 0
+            and self.limit_train_batches > 0.0
+            and isinstance(self.limit_train_batches, float)
+            and orig_train_batches != float("inf")
+        ):
+            min_percentage = 1.0 / orig_train_batches
+            raise MisconfigurationException(
+                f"You requested to check {self.limit_train_batches} of the `train_dataloader` but"
+                f" {self.limit_train_batches} * {orig_train_batches} < 1. Please increase the"
```

Comment:
```suggestion
                f" {self.limit_train_batches} * {orig_train_batches} = 1. Please increase the"
```
of can it be negative?

Reply:
should be `< 1` no? the error is thrown for cases `[0, 1)`. Can't be negative.

Reply:
I guess it can be between 0 and 1, since `limit_train_batches` can be a float